pennylane_rigetti/qc.py

```diff
@@ -223,56 +249,169 @@ def apply_parametric_operations(self, operations):
 
             # Prepare for parametric compilation
             par = []
-            for param in operation.data:
-                if getattr(param, "requires_grad", False) and operation.name != "BasisState":
-                    # Using the idx for trainable parameter objects to specify the
-                    # corresponding symbolic parameter
-                    parameter_string = "theta" + str(id(param))
-
-                    if parameter_string not in self._parameter_reference_map:
-                        # Create a new PyQuil memory reference and store it in the
-                        # parameter reference map if it was not done so already
-                        current_ref = self.prog.declare(parameter_string, "REAL")
-                        self._parameter_reference_map[parameter_string] = current_ref
-
-                    # Store the numeric value bound to the symbolic parameter
-                    self._parameter_map[parameter_string] = [param.unwrap()]
-
-                    # Appending the parameter reference to the parameters
-                    # of the corresponding operation
-                    par.append(self._parameter_reference_map[parameter_string])
-                else:
-                    par.append(param)
+            if operation.batch_size is not None:
+                parameter_string = f"theta{i}"
+                if parameter_string not in self._parameter_reference_map:
+                    # Create a new PyQuil memory reference and store it in the
+                    # parameter reference map if it was not done so already
+                    current_ref = self.prog.declare(parameter_string, "REAL")
+                    self._parameter_reference_map[parameter_string] = current_ref
+
+                # Store the values bound to the symbolic parameter
+                self._batched_parameter_map[parameter_string] = operation.data[0]
```

Comment:
What if it's a `Rot` gate?  They might have more than one parameter.  Maybe just only allow parameters with one parameter here?